src/traces/icicle/plot.js

```diff
@@ -0,0 +1,625 @@
+'use strict';
```

Comment:
This file still looks very similar to `src/traces/treemap/plot.js`.
On a dev branch could you please investigate the possibility of reusing / adapting `treemap/plot`?

Reply:
Sure, I'll look into this.

Reply:
This one is not resolved.
What we need is to rename current plot.js file in `treemap` to `draw.js` and pass in the `type` as well as the required `drawDescendants` function from `icicle` and `treemap` "plot" methods.

Reply:
now resolved by https://github.com/plotly/plotly.js/pull/5546/commits/eaad10dd82061e196914873f0c350ae36caca88b